test_common/Test_Common_ArithTraits.hpp

```diff
@@ -371,6 +371,19 @@ class ArithTraitsTesterBase {
       success = 0;
     }
 
+    if (AT::hasInfinity) {
+      if (! AT::isInf (AT::infinity())) {
+        out << "AT::isInf (inf) != true" << endl;
+        success = 0;
+      }
+    } else {
```

Comment:
I'd rather have had a static assertion
```C++
    static_assert(
      AT::hasInfinity && (!AT::is_integer && !AT::is_complex) ||
        !AT::hasInfinity && (AT::is_integer || !AT::is_complex),
      "real floating-point type cannot represent positive infinity value");
```
rather than that `else` branch but it is not possible because `is_complex` and `is_integer` are not `constexpr`.

Reply:
@dalg24 The main thing hindering `constexpr` here was CUDA not permitting `constexpr` values to be marked `__device__`.  I think that has been fixed, though you may have to select on the CUDA version in order to use `constexpr` without warnings, and/or use a special `nvcc` flag.